clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -204,23 +204,14 @@ public synchronized void ensureCoordinatorReady() {
                 time.sleep(retryBackoffMs);
             }
         }
+        // Since coordinator is set in a synchronized block in the group coordinator response callback,
+        // the future associated with the cooordinator request can be safely cleared here.
+        findCoordinatorFuture = null;
```

Comment:
Looking at this a little more, I think @ijuma is right that the shadowing is unnecessary.